include/net/http_client.h

```diff
@@ -0,0 +1,291 @@
+/** @file
+ * @brief HTTP client high-level api implementation for Zephyr.
+ */
+
+/*
+ * Copyright (c) 2019 Unisoc Technologies INC.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_HTTP_CLIENT_H_
+#define ZEPHYR_INCLUDE_HTTP_CLIENT_H_
+
+#include <net/http_parser.h>
+
+#ifdef CONFIG_HTTPS
+	#define CONFIG_NET_SOCKETS_SOCKOPT_TLS
+#endif
+
+#if !defined(ZEPHYR_USER_AGENT)
+#define ZEPHYR_USER_AGENT "Zephyr"
+#endif
+
+#if !defined(HTTP_PROTOCOL)
+#define HTTP_PROTOCOL	   "HTTP/1.1"
+#endif
+
+#define HTTP_CRLF "\r\n"
+#define HTTP_STATUS_OK			"OK"
+#define HTTP_PARTIAL_CONTENT	"Partial Content"
+
+#define HOST_NAME_MAX_SIZE		32
+#define HTTP_REQUEST_MAX_SIZE	1024
+#define HTTP_RESPONSE_MAX_SIZE	4096
+#define HTTP_STATUS_STR_SIZE	16
+
+#if defined(CONFIG_NET_SOCKETS_SOCKOPT_TLS)
+	#define HTTP_PORT "443"
+#else
+	#define HTTP_PORT "80"
+#endif
+
+struct http_ctx;
+
+/* HTTP header fields struct */
+struct http_field_value {
+	/** Field name, this variable will point to the beginning of the string
+	 *  containing the HTTP field name
+	 */
+	const char *key;
+
+	/** Value, this variable will point to the beginning of the string
+	 *  containing the field value
+	 */
+	const char *value;
+
+	/** Length of the field name */
+	u16_t key_len;
+
+	/** Length of the field value */
+	u16_t value_len;
+};
+
+/* Is there more data to come */
+enum http_final_call {
+	HTTP_DATA_MORE = 0,
+	HTTP_DATA_FINAL = 1,
+};
+
+/* http header filed and value */
+enum http_header_field_val {
+	HTTP_HEADER_FIELD = 0,
+	HTTP_HEADER_VALUE = 1,
+};
+
+/**
+ * @typedef http_response_cb_t
+ * @brief Callback used when a response has been received from peer.
+ *
+ * @param ctx HTTP context.
+ * @param data Received body point
+ * @param received body length
+ * @param final_data Does this data buffer contain all the data or
+ * is there still more data to come.
+ * @param user_data A valid pointer on some user data or NULL
+ */
+typedef void (*http_response_cb_t)(struct http_ctx *ctx,
+					const char *body,
```

Comment:
indent the parameter so it goes below the last '(' + 1 space.

Apply that everywhere